### ISSUE TITLE
Data is rounded

### ISSUE THREAD

**Issue by arnoldpistorius** [fsaris/home-assistant-zonneplan-one#94]
Hi,

Since a few days I see some strange behaviour within the energy page of Home Assistant. It suddenly detected negative solar yield and since then it rounds all readings. In the graph below you see the negative reading and then all readings are rounded to 1.0:

![image](https://github.com/fsaris/home-assistant-zonneplan-one/assets/6072267/b34a47ff-792a-4c31-a918-03b99f1b613c)

Never saw a negative reading again, but I only see 1.0 readings now. I didn't do any updates of HA or this addon before it started to happen. I tried updating the addon, but still provides me only with 1.0 readings.

The sensor values just seem to be correct. I can imagine some faulty reading caused the HA database to be corrupted, but I know too little about it.

Do you have any idea what the issue can be?

**Comment by rodavrow** [comment]
For me this started in exactly the same way on the 16th. It makes me wonder if they have changed something with their API?

**Comment by rddoove** [comment]
I have the sam issues. All graph are rounded and outcome too
![Schermafbeelding 2024-01-19 om 18 24 18 jpeg](https://github.com/fsaris/home-assistant-zonneplan-one/assets/26503612/41e4d75a-b011-48fa-a89a-8be53a3ec03e)

**Comment by fsaris** [comment]
Do you also see this in the stats of the sensor `Zonneplan yield today`?  

@arnoldpistorius is your graph from the energy dashboard?

The values of the sensor have at least a 2 digit precision. The the 1.0 value seems odd.

**Comment by arnoldpistorius** [comment]
Yes my graph was from the energy dashboard.

Here's a screenshot of my zonneplan entities:

![image](https://github.com/fsaris/home-assistant-zonneplan-one/assets/6072267/e6285202-9944-4fa9-8306-34bdf76f6a6f)


It seems the zonneplan total yield is rounded as well. Maybe that could be related..

![image](https://github.com/fsaris/home-assistant-zonneplan-one/assets/6072267/3447cfac-5311-42e2-806a-8b69de0c8eac)

**Comment by fsaris** [comment]
Maybe use yield today. That is updated more often and with more precision.

Total is currently not updated that often see https://github.com/fsaris/home-assistant-zonneplan-one/pull/91

**Comment by rddoove** [comment]
Indeed that did the trick. I used yield today in the energy dashboard

**Comment by arnoldpistorius** [comment]
The yield today sensor works indeed. However, my energy history is gone now ¯\_(ツ)_/¯.

**Comment by fsaris** [comment]
found the data in the info we already fetch every 5 minutes. please test `main` version.

**Comment by martijnrusschen** [comment]
Yes this fixed the issue! Can confirm the last measured value is now updated every 5 min again. See screenshot below with the before/after. 

![Screenshot 2024-02-19 at 08 59 42@2x](https://github.com/fsaris/home-assistant-zonneplan-one/assets/1412392/d4828f5d-2c5c-4a24-83c3-2da4cc5267f6)
